includes/class-wc-customer-download.php

```diff
@@ -258,14 +273,56 @@ public function set_access_expires( $date = null ) {
 	}
 
 	/**
-	 * Get download_count.
+	 * Set download_count.
 	 *
 	 * @param int $value
 	 */
 	public function set_download_count( $value ) {
 		$this->set_prop( 'download_count', absint( $value ) );
 	}
 
+	/**
+	 * Track a download on this permission.
+	 *
+	 * @since 3.3.0
+	 * @param int user_id Id of the user performing the download
+	 * @param string user_ip_address IP Address of the user performing the download
+	 */
+	public function track_download( $user_id = null, $user_ip_address = null ) {
+		// Must have a permission_id to track download log
+		if ( ! ( $this->get_id() > 0 ) ) {
+			throw new Exception( __( 'Invalid permission ID.', 'woocommerce' ) );
+		}
+
+		// Increment download count on parameter
+		$count = $this->get_download_count();
+		$this->set_download_count( $count + 1 );
```

Comment:
I wonder if doing this with PHP may cause issues if downloads are made in quick succession. Would it be more reliable to do this in SQL?